### ISSUE TITLE
Kafka engine information in systems table/logging level kafka engine

### ISSUE THREAD

**Issue by yctn** [ClickHouse/ClickHouse#5128]
It would be very usefull to provide kafka information in a system table.
it then can be queryed for monitoring of the kafka ingestion.
things suchs as commited offsets. topic's amount of partitions etc.

secondly, it would be good to provide a way/config settings to set the logging level of librdkafka currently its set to default so the clickhouse-server.log receives huge spamming of all the librdkafka debug information.

**Comment by abyss7** [comment]
About logging. In recent versions the Kafka debug level is set to nothing by default - and even if you use older version, you can setup the "kafka" section inside the server config xml.

**Comment by filimonov** [comment]
I've found yesterday that rdkafka have statistics API:
https://github.com/edenhill/librdkafka/blob/master/STATISTICS.md 

Also cppkafka has wrapper for that:  
https://github.com/mfontanini/cppkafka/blob/29cb02b756a79a9f77c36e95cbcf799d3e3edfc2/src/configuration.cpp#L90

That can be exposed in something like system.kafka_stats.

**Comment by filimonov** [comment]
The situation with logs improved in https://github.com/ClickHouse/ClickHouse/pull/10983 (will be released in 20.5) https://github.com/ClickHouse/ClickHouse/pull/11261 

Stats & introspection is in TODO

**Comment by filimonov** [comment]
proposal:

new system table: system.kafka_consumers
- database String
- table String
- consumer_id (number? or string? - single clickhouse table can have many consumers)
- assignments.topic Array(String)
- assignments.partition_id Array(String)
- assignments.current_offset Array(String)
- last_exception_time DateTime
- last_exception String
- last_poll_time DateTime
- num_messages_read UInt64
- last_commit_time DateTime
- num_commits UInt64 
- last_rebalance_time DateTime
- num_rebalance_revocations UInt64
- num_rebalance_assignments UInt64
- is_currently_used UInt8
- librdkafka_statistics String (json - https://github.com/edenhill/librdkafka/blob/master/STATISTICS.md )

**Comment by morsapaes** [comment]
The system catalog table was added in v23.8. Before closing the issue: @antaljanosbenjamin, since the table engine can also be used to produce messages, should we consider a more generic (or dedicated) version of this system table for producer-side metrics?

**Comment by antaljanosbenjamin** [comment]
> The system catalog table was added in v23.8

Just to make sure we are talking about the same thing: you mean `system.kafka_consumers`, right? If yes, then it is correct. However **it is not supported by Kafka2**.

> since the table engine can also be used to produce messages, should we consider a more generic (or dedicated) version of this system table for producer-side metrics?

That is a tricky question. By my heart I would say it doesn't make sense to in the same way, because producers are short lived entities. We can create a similar system tables for producer side metrics, but in my opinion that shouldn't be tied to any specific producer, but kafka tables and the stats should be aggregated in some way. I can think of what information can be useful if we think we need something like that.

**Comment by morsapaes** [comment]
> Just to make sure we are talking about the same thing: you mean system.kafka_consumers, right? If yes, then it is correct. However it is not supported by Kafka2.

Correct. I meant this for v1 since that's what this issue referred to. Let's track what we discussed live for statistics reporting in v2 and also global table statistics in a different issue.
